deploy-service/common/src/main/java/com/pinterest/deployservice/handler/DeployHandler.java

```diff
@@ -395,6 +395,14 @@ private void validateBuild(EnvironBean envBean, String buildId) throws Exception
             throw new DeployInternalException("Build name (%s) does not match stage config (%s).",
                     buildBean.getBuild_name(), envBean.getBuild_name());
         }
+        
+        if(envBean.getStage_type() != EnvType.DEV &&
+                buildAllowlist != null && !buildAllowlist.trusted(buildBean.getArtifact_url())
+                && buildBean.getScm_branch() != null
+                && !buildBean.getScm_branch().toLowerCase().startsWith(PRIVATE_BUILD_SCM_BRANCH)) {
+            buildBean.setScm_branch(PRIVATE_BUILD_SCM_BRANCH);
+        }
+
         // only allow a non-private deploy if the build is from a trusted artifact url
         if(envBean.getEnsure_trusted_build() && !buildBean.getScm_branch().equals("private") &&
```

Comment:
Let's make a more complete change.
1. Create a new method `isPrivateBuild(BuildBean)` and returns `buildBean.getScm_branch() != null
                && !buildBean.getScm_branch().toLowerCase().startsWith(PRIVATE_BUILD_SCM_BRANCH)`
2. Replace checks like `!buildBean.getScm_branch().equals("private")` to the new method. 

So we have a consistent way of telling if a build is a private build.

Reply:
I'd like to see test coverage for these changes.